source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -8,31 +8,54 @@ namespace RBAC {
 
 RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     const envoy::config::filter::http::rbac::v2::RBAC& config, bool disabled)
-    : engine_disabled_(disabled),
-      allowed_if_matched_(disabled ||
-                          config.rules().action() ==
-                              envoy::config::rbac::v2alpha::RBAC_Action::RBAC_Action_ALLOW) {
+    : engine_disabled_(disabled), config_(config) {
+
   if (disabled) {
```

Comment:
This could also be removed?

Reply:
as mentioned in above comment, keep this for now because of per-route config; it could be removed if we could decide disable from RBACPerRoute.RBAC field(which this PR uses in other places). I'm not familiar about the RBACPerRoute use case, wait @rodaine to comment whether it's safe to remove disable field from RBACPerRoute proto.

Reply:
I mean the if statement.

Reply:
oh I misunderstand, done.